### PR TITLE
Collision

### PR DESCRIPTION
**Pull request by Suedie**

### CODE REVIEW

File: core/src/Model/Entity.java [hunk 1]
```diff
@@ -1,47 +1,63 @@
...
+
+    //TODO rework movement, remove has-dependency on world
+    public Entity(float x, float y, float height, float width, float speed,float health, World<Entity> world) {
         this.x = x;
         this.y = y;
         this.height = height;
         this.width = width;
         this.speed = speed;
         this.health = health;
+        setWorld(world);
```
Comment (rojtekin) [inline core/src/Model/Entity.java:40]: skapar detta inte cirkulär beroende?

File: core/src/Model/Entity.java [hunk 2]
```diff
@@ -1,47 +1,63 @@
...
-
-    public Entity(int x, int y, int height, int width, int speed,float health) {
+    private float x;
+    private float y;
+    private float height;
+    private float width;
+    private float speed;
+    private float health;
+    private Item<Entity> boundingbox;
+    private World<Entity> world;
```
Comment (pkasinski) [inline core/src/Model/Entity.java:20]: Does each entity need a world?

**Review by rojtekin — APPROVED** [review]
LGTM, just worried about cirkular dependecy on world thing in Entity class getworld setworld functions

File: core/src/Model/StaticObjectEntity.java [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package Model;
+
+import com.dongbat.jbump.World;
+
+public class StaticObjectEntity extends Entity {
```
Comment (pkasinski) [inline core/src/Model/StaticObjectEntity.java]: Documentation could be added here.

**Review by pkasinski — APPROVED** [review]
Looks good!

File: core/src/Model/StaticObjectEntity.java (hunk 3 above)
Comment (Suedie) [reply core/src/Model/StaticObjectEntity.java]: Javadoc added

File: core/src/Model/Entity.java (hunk 1 above)
Comment (Suedie) [reply core/src/Model/Entity.java:40]: Tar upp det med Pelle

File: core/src/Model/Entity.java (hunk 2 above)
Comment (Suedie) [reply core/src/Model/Entity.java:20]: Will take it up with Pelle

**Pull request merged**